src/lib/storage/chunk.cpp

```diff
@@ -15,26 +15,26 @@
 namespace opossum {
 
 void Chunk::add_segment(std::shared_ptr<BaseSegment> segment) {
-  // Implementation goes here
+  Assert(size() == 0 || segment->size() == size(), "Segment needs to have same number of rows as chunk");
+  _segments.push_back(segment);
 }
 
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  DebugAssert(values.size() == column_count(), "New values need same number of values as chunk columns");
+  for (auto i = 0ul, end = values.size(); i < end; i++) {
+    _segments[i]->append(values[i]);
+  }
 }
 
-std::shared_ptr<BaseSegment> Chunk::get_segment(ColumnID column_id) const {
-  // Implementation goes here
-  return nullptr;
-}
+std::shared_ptr<BaseSegment> Chunk::get_segment(ColumnID column_id) const { return _segments.at(column_id); }
 
-ColumnCount Chunk::column_count() const {
-  // Implementation goes here
-  return ColumnCount{0};
-}
+ColumnCount Chunk::column_count() const { return ColumnCount{_segments.size()}; }
```

Comment:
Compiler throws an error: narrow conversion of long unsigned int to short unsigned int - 
```suggestion
ColumnCount Chunk::column_count() const { return ColumnCount{static_cast<ColumnCount::base_type>(_segments.size())}; }
```